src/data.rs

```diff
@@ -0,0 +1,1190 @@
+    unsafe { transmute(l) }
+  }
+}
+
+/// An instance of the built-in Set constructor (ECMA-262, 6th Edition, 23.2.1).
+#[repr(C)]
+pub struct Set(Opaque);
+impl<'sc> From<Local<'sc, Set>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, Set>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, Set>> for Local<'sc, Value> {
+  fn from(l: Local<'sc, Set>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, Set>> for Local<'sc, Object> {
+  fn from(l: Local<'sc, Set>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+
+/// An instance of the built-in SharedArrayBuffer constructor.
+/// This API is experimental and may change significantly.
+#[repr(C)]
+pub struct SharedArrayBuffer(Opaque);
+impl<'sc> From<Local<'sc, SharedArrayBuffer>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, SharedArrayBuffer>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, SharedArrayBuffer>> for Local<'sc, Value> {
+  fn from(l: Local<'sc, SharedArrayBuffer>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, SharedArrayBuffer>> for Local<'sc, Object> {
+  fn from(l: Local<'sc, SharedArrayBuffer>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+
+/// A String object (ECMA-262, 4.3.18).
+#[repr(C)]
+pub struct StringObject(Opaque);
+impl<'sc> From<Local<'sc, StringObject>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, StringObject>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, StringObject>> for Local<'sc, Value> {
+  fn from(l: Local<'sc, StringObject>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, StringObject>> for Local<'sc, Object> {
+  fn from(l: Local<'sc, StringObject>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+
+/// A Symbol object (ECMA-262 edition 6).
+#[repr(C)]
+pub struct SymbolObject(Opaque);
+impl<'sc> From<Local<'sc, SymbolObject>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, SymbolObject>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, SymbolObject>> for Local<'sc, Value> {
+  fn from(l: Local<'sc, SymbolObject>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, SymbolObject>> for Local<'sc, Object> {
+  fn from(l: Local<'sc, SymbolObject>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+
+#[repr(C)]
+pub struct WasmModuleObject(Opaque);
+impl<'sc> From<Local<'sc, WasmModuleObject>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, WasmModuleObject>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, WasmModuleObject>> for Local<'sc, Value> {
+  fn from(l: Local<'sc, WasmModuleObject>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, WasmModuleObject>> for Local<'sc, Object> {
+  fn from(l: Local<'sc, WasmModuleObject>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+
+/// The superclass of primitive values.  See ECMA-262 4.3.2.
+#[repr(C)]
+pub struct Primitive(Opaque);
```

Comment:
It would be nice for the struct declarations to live next to their corresponding methods - e.g. Primitive gets declared in src/primitive.rs

Reply:
I don't understand why they all got moved to this data.rs file - or why it's called data.rs ?

Reply:
https://github.com/denoland/rusty_v8/blob/d128ddf1c76059e3eaa5cab82340b5450e89a2e2/src/data.rs#L7

I'm open to a better name.

They all got moved because that made auto generating easier.
I expect to rerun the generator some more times, e.g. to automatically add Deref implementations.

Reply:
I think it would be better to leave the struct declarations in their own file.

This file might be called from.rs ?